### PR TITLE
Organize export / import

### PR DESCRIPTION
**Pull request by junha1**

### CODE REVIEW

File: remote-trait-object/src/service/export_import.rs [hunk 1]
```diff
@@ -14,77 +14,68 @@
...
+use std::sync::Arc;
 
 // These traits are associated with some specific service trait.
 // These tratis will be implement by `dyn ServiceTrait` where `T = dyn ServiceTrait` as well.
 // Macro will implement this trait with the target(expanding) service trait.
 
-pub trait ExportServiceBox<T: ?Sized + Service> {
-    fn export(port: Weak<dyn Port>, object: Box<T>) -> HandleToExchange;
+/// Unused T is for avoiding violation of the orphan rule
+/// P will be local type for the crate, and that makes it possible to
```
Comment (majecty) [inline remote-trait-object/src/service/export_import.rs:27]: What is the `P`?

File: remote-trait-object/src/service/serde_support.rs [hunk 2]
```diff
@@ -20,65 +20,32 @@ use parking_lot::RwLock;
...
-        self.value.take().unwrap()
-    }
-
-    pub fn unwrap(self) -> Arc<RwLock<T>> {
-        self.value.take().unwrap()
-    }
-}
+pub type SBox<T> = ServicePointer<Box<T>, T>;
+pub type SArc<T> = ServicePointer<Arc<T>, T>;
+pub type SRwLock<T> = ServicePointer<Arc<RwLock<T>>, T>;
```
Comment (majecty) [inline remote-trait-object/src/service/serde_support.rs:48]: I think the previous version is easier to understand.
Why did you change this code?

File: remote-trait-object/src/service/export_import.rs [hunk 3]
```diff
@@ -14,77 +14,68 @@
...
-
-pub trait ImportServiceArc<T: ?Sized + Service> {
-    fn import(port: Weak<dyn Port>, handle: HandleToExchange) -> Arc<T>;
-}
-
-pub trait ImportServiceRwLock<T: ?Sized + Service> {
-    fn import(port: Weak<dyn Port>, handle: HandleToExchange) -> Arc<RwLock<T>>;
+/// Unused T is for avoiding violation of the orphan rule, like `ToDispatcher`
+pub trait ToRemote<T: ?Sized + Service> {
+    fn to_remote(port: Weak<dyn Port>, handle: HandleToExchange) -> Self;
```
Comment (majecty) [inline remote-trait-object/src/service/export_import.rs:37]: I don't understand why we need `ToDispatcher` and `ToRemote`. Could you explain it to me?

File: remote-trait-object/src/service/serde_support.rs [hunk 4]
```diff
@@ -101,74 +101,81 @@ impl<'de, P: ?Sized + Service, T: ToRemote<P>> Deserialize<'de> for ServicePoint
...
+                unimplemented!()
             }
-        }
-        impl Foo for FooImpl {
-            fn get_handle_to_exchange(&self) -> HandleToExchange {
-                self.handle_to_exchange
+
+            fn register(&self, _service_object: Arc<dyn Dispatch>) -> HandleToExchange {
+                self.count.fetch_add(1, Ordering::SeqCst);
+                HandleToExchange(123)
```
Comment (majecty) [inline remote-trait-object/src/service/serde_support.rs:123]: I don't understand why you created this commit: "Fix the test which depends on export detail". Could you explain it to me?

File: remote-trait-object/src/service/export_import.rs (hunk 1 above)
Comment (junha1) [reply remote-trait-object/src/service/export_import.rs:27]: @majecty It is typo of `T`. I'll fix
